handler/src/main/java/io/netty/handler/ipfilter/RuleBasedIpFilter.java

```diff
@@ -22,36 +22,71 @@
 
 import java.net.InetSocketAddress;
 import java.net.SocketAddress;
+import java.util.ArrayList;
+import java.util.List;
 
 /**
+ * <p>
  * This class allows one to filter new {@link Channel}s based on the
  * {@link IpFilterRule}s passed to its constructor. If no rules are provided, all connections
  * will be accepted.
+ * </p>
  *
+ * <p>
  * If you would like to explicitly take action on rejected {@link Channel}s, you should override
  * {@link AbstractRemoteAddressFilter#channelRejected(ChannelHandlerContext, SocketAddress)}.
+ * </p>
+ *
+ * <p> You can also consider using {@link IpSubnetFilter} which uses Binary Search Algorithm
+ * for super fast IP Address filtering. </p>
```

Comment:
nit: I would remove "super fast" here... what does this even mean ?

Reply:
`RuleBasedIpFilter` uses a for-each loop but `IpSubnetFilter` uses Binary Search which is a lot faster and should be considered over `RuleBasedIpFilter` that's why I added "Super fast" there.

Should I remove?

Reply:
Maybe just use something like:

`Consider using {@link IpSubnetFilter} for better performance while not as general purpose as this filter`.